src/test/unit/bisect.test.js

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import { bisectionLeft, bisectionRight } from '../../utils/bisect';
+
+describe('bisectionRight', function() {
+  const array = [0, 1, 2, 3, 3, 3, 3, 3, 5, 6, 7, 8, 9];
+  // The new element to be inserted into this array is referred to as x
+  it('returns index of the first number greater than x', function() {
+    expect(bisectionRight(array, 4)).toBe(8);
+    expect(bisectionRight(array, 3)).toBe(8);
+  });
+
+  it('returns index of the first number greater than x, occuring after low', function() {
+    expect(bisectionRight(array, 4, 8)).toBe(8);
+    expect(bisectionRight(array, 3, 9)).toBe(9);
+    expect(bisectionLeft(array, 2, -100)).toBe(2);
```

Comment:
Looking at this test, I wonder if we shouldn't guard against specifying bad low and high values.
For example, at the start of the function:
```js
if (low < 0) {
  // throw error
 }
```
The python implementation throws an error for this case: https://github.com/python/cpython/blob/3fc7080220b8dd2e1b067b3224879133d895ea80/Lib/bisect.py#L26-L27
But doesn't check other cases. Then maybe we should do the same, with a `TypeError` (this is the equivalent in JavaScript to the Python's `ValueError`), what do you think?

Reply:
I agree we should handle bad low and high values, because especially in case when low is greater than the array length or both low and high are greater, we are getting absurd values, trying to refer to an element with that large index. 
I believe both low and high should be in the range of the array and anything apart from that should throw an error.
Thanks for the link to the python implementation, I will check it out and add the error handling aspect to this.